liblocation/src/main/java/com/mapbox/android/core/location/LocationPendingIntentProvider.java

```diff
@@ -0,0 +1,49 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+
+import java.util.concurrent.CopyOnWriteArrayList;
+
+class LocationPendingIntentProvider {
+
+  private static final int REQUEST_CODE = 0;
+  private final LocationPendingIntent locationPendingIntent;
+
+  LocationPendingIntentProvider(Context context, SdkChecker sdkChecker,
+                                CopyOnWriteArrayList<LocationEngineListener> locationEngineListeners) {
+    locationPendingIntent = buildIntent(context, sdkChecker, locationEngineListeners);
+  }
+
+  LocationPendingIntent intent() {
+    return locationPendingIntent;
+  }
+
+  private LocationPendingIntent buildIntent(Context context, SdkChecker sdkChecker,
```

Comment:
It feels like this method can also be static, I can't see we would want to hold a reference to the `LocationPendingIntentProvider`.